### DISCUSSION TITLE
[Question] How to support images of various resolutions?

### DISCUSSION THREAD

**Discussion by RukshanJS** [flbulgarelli/headbreaker#36] [Q&A] [answered]
I have a use case where I want to use images dynamically. Also, the number of tiles (puzzle) should be able to change for each image. (4 pieces, 16 pieces...) How would one approach this problem?

With the current implementation (as per docs), some images are cut-off then I use the code.
```
autogen.autogenerate({
    horizontalPiecesCount: 6,
    verticalPiecesCount: 5
  });
```

**Comment by flbulgarelli** [comment, marked as answer]
Hi! If you want to dynamically change images - or their resolution - you should probably try just dropping the canvas and rendering it again with the new image.

Changing image on an already shown puzzle may be possible, but you might have to hack it a little, since it is not supported out of the box.

Also, by doing that you will be able to change puzzle's size and pieces. Please have a look at https://github.com/flbulgarelli/headbreaker/issues/24, which is very close to your question.

**Comment by flbulgarelli** [comment]
PS:  

> With the current implementation (as per docs), some images are cut-off then I use the code.

I guess you are using [this example](https://flbulgarelli.github.io/headbreaker/#autogenerated), right? Have you tried using `canvas.adjustImagesToPuzzleHeight()` or `canvas.adjustImagesToPuzzleWidth()` methods, too?
